x-pack/plugins/aiops/server/routes/log_rate_analysis/queries/fetch_categories.ts

```diff
@@ -28,30 +28,68 @@ import { isRequestAbortedError } from '../../../lib/is_request_aborted_error';
 
 import { getQueryWithParams } from './get_query_with_params';
 
+// Filter that includes docs from both the baseline and deviation time range.
+export const getBaselineOrDeviationFilter = (
+  params: AiopsLogRateAnalysisSchema
+): estypes.QueryDslQueryContainer => {
+  return {
+    bool: {
+      should: [
+        {
+          range: {
+            [params.timeFieldName]: {
+              gte: params.baselineMin,
+              lte: params.baselineMax,
+              format: 'epoch_millis',
+            },
+          },
+        },
+        {
+          range: {
+            [params.timeFieldName]: {
+              gte: params.deviationMin,
+              lte: params.deviationMax,
+              format: 'epoch_millis',
+            },
+          },
+        },
+      ],
+    },
+  };
+};
+
 export const getCategoryRequest = (
   params: AiopsLogRateAnalysisSchema,
   fieldName: string,
-  from: number | undefined,
-  to: number | undefined,
-  filter: estypes.QueryDslQueryContainer,
   { wrap }: RandomSamplerWrapper
 ): estypes.SearchRequest => {
   const { index, timeFieldName } = params;
+
   const query = getQueryWithParams({
-    params,
+    // Passing in an empty string for the time field name will avoid
+    // adding any range queries. We're enforcing this here since this
+    // is covered by the filter which will match docs in both baseline
+    // and deviation time range via `getBaselineOrDeviationFilter`.
+    params: { ...params, timeFieldName: '' },
```

Comment:
It's not very clear within the function `getQueryWithParams` that this is happening when timeFieldName == '' (since the logic is in `getFilters`). Suggestion to add a comment to document the function that this is the case. Also, is there a reason we don't we use an empty string instead of undefined or null?

Reply:
I'm assuming the empty string is due to `AiopsLogRateAnalysisSchema` ... I think we can extend the time to support nullable timeFieldName here for clarity 🤔  Although this suggested change is not a blocker.

Reply:
You're right this was a bit intransparent. Yes, `timeFieldName` is required for `AiopsLogRateAnalysisSchema` and should stay like that. `get_filter.ts/getFilters` was a too generic name, I'm not sure this included other stuff in earlier iterations but the only thing that was left in there was the check for an empty `timeFieldName` and the code to generate the range query. Now I got rid of that whole file and just kept `getRangeQuery`. `getQueryWithParams` now has a boolean flag `skipRangeQuery`, based on that we'll add the range query based on `start/end/timeFieldName`. Hope that makes it more clear. Updated in 84647fa.